source.lua

```diff
@@ -0,0 +1,145 @@
+--@ module = true
+-- wip
+
+local repeatUtil = require('repeat-util')
+
+liquidSources = liquidSources or {}
+
+local sourceId = 'liquidSources'
+
+function isFlowPassable(pos)
+    local tiletype = dfhack.maps.getTileType(pos.x, pos.y, pos.z)
+    local titletypeAttrs = df.tiletype.attrs[tiletype]
+    local shape = titletypeAttrs.shape
+    local tiletypeShapeAttrs = df.tiletype_shape.attrs[shape]
+    return tiletypeShapeAttrs.passable_flow
+end
+
+function AddLiquidSource(pos, liquid, amount)
+    table.insert(liquidSources, {
+        liquid = liquid,
+        amount = amount,
+        pos = {
+            x = pos.x,
+            y = pos.y,
+            z = pos.z,
+        },
+    })
+
+    repeatUtil.scheduleEvery(sourceId, 12, 'ticks', function()
+        if next(liquidSources) == nil then
+            repeatUtil.cancel(sourceId)
+        else
+            for _, v in pairs(liquidSources) do
+                local block = dfhack.maps.getTileBlock(v.pos.x, v.pos.y, v.pos.z)
+                local x = v.pos.x
+                local y = v.pos.y
+                if block and isFlowPassable(v.pos) then
+                    local isMagma = v.liquid == 'magma'
+
+                    local flow = block.designation[x%16][y%16].flow_size
```

Comment:
I believe you can replace this with the following, then you can simplify lines 48-50 to use tileFlags as well

```suggestion
                    local tileFlags = dfhack.maps.getTileFlags(v.pos)
                    local flow = tileFlags.flow_size
```

Reply:
This is how every existing script does it, I assume the API is relatively new. I'll change it over.